terraform/README.md

```diff
@@ -0,0 +1,65 @@
+# Terraform module for `cos-registration-server`
+
+This is a Terraform module facilitating the deployment of the `cos-registration-server` charm,
+using the [Terraform Juju provider](https://github.com/juju/terraform-provider-juju/).
+For more information,
+refer to the provider [documentation](https://registry.terraform.io/providers/juju/juju/latest/docs).
+
+## Requirements
+
+This module requires a Juju K8s model to be available.
+Refer to the [usage section](#usage) below for more details.
+
+## Usage
+
+Users should ensure that Terraform is aware of the `juju_model` dependency of the charm module.
+
+To deploy this module with its needed dependency, you can run:
+
+```bash
+terraform apply -var="model=<MODEL_NAME>"
+```
+
+<!-- BEGIN_TF_DOCS -->
+## Requirements
+
+| Name | Version |
+|------|---------|
+| terraform | >= 1.5 |
+| juju | ~> 0.19.0 |
```

Comment:
is this the juju provider version or actual Juju?
Asking because the provider version is also given below.

Reply:
This is the juju **provider** version.
Note that the whole content between the `BEGIN_TF_DOCS` is automatically generated by terraform-docs.